code/modules/mob/living/carbon/xenomorph/castes/king/abilities_conqueror.dm

```diff
@@ -0,0 +1,979 @@
+		disable_ability()
+	return ..()
+
+/datum/action/ability/xeno_action/conqueror_obliteration/on_cooldown_finish()
+	. = ..()
+	xeno_owner.playsound_local(xeno_owner, 'sound/effects/alien/new_larva.ogg', 35, 0)
+	xeno_owner.balloon_alert(xeno_owner, "[initial(name)] ready")
+
+/datum/action/ability/xeno_action/conqueror_obliteration/can_use_action(...)
+	. = ..()
+	if(!.)
+		return
+	if(!xeno_owner.canmove)
+		return FALSE
+	return TRUE
+
+/datum/action/ability/xeno_action/conqueror_obliteration/keybind_activation()
+	if(can_use_action())
+		if(toggled)
+			INVOKE_ASYNC(src, PROC_REF(check_ability), TRUE)
+		else
+			INVOKE_ASYNC(src, PROC_REF(enable_ability), TRUE)
+	return COMSIG_KB_ACTIVATED
+
+/datum/action/ability/xeno_action/conqueror_obliteration/action_activate()
+	. = ..()
+	if(toggled)
+		check_ability()
+		return
+	enable_ability()
+
+/datum/action/ability/xeno_action/conqueror_obliteration/alternate_action_activate()
+	. = ..()
+	if(toggled)
+		check_ability()
+		return
+	enable_ability()
+
+/datum/action/ability/xeno_action/conqueror_obliteration/clean_action()
+	if(toggled)
+		disable_ability()
+	return ..()
+
+/datum/action/ability/xeno_action/conqueror_obliteration/process()
+	if(!can_use_action(TRUE))
+		disable_ability()
+		return
+	if(ability_range < CONQUEROR_OBLITERATION_MAX_RANGE)
+		ability_range++
+	if(!particle_holder)
+		return
+	var/particle_scale = 0.2 + (ability_range * 0.4)
+	particle_holder.particles.scale = list(particle_scale, particle_scale)
+	animate(particle_holder, time = 0, alpha = initial(particle_holder.alpha), color = initial(particle_holder.color), flags = ANIMATION_END_NOW)
+	animate(time = 1 SECONDS, alpha = 0, color = "#000000")
+
+/// Enables the abiity. If activated via the keybind, releasing it will move on to the next step.
+/datum/action/ability/xeno_action/conqueror_obliteration/proc/enable_ability(keybind = FALSE)
+	toggled = TRUE
+	set_toggle(TRUE)
+	xeno_owner.add_movespeed_modifier(MOVESPEED_ID_CONQUEROR_OBLITERATION, TRUE, 0, NONE, TRUE, CONQUEROR_OBLITERATION_SPEED_MODIFIER)
+	ADD_TRAIT(xeno_owner, TRAIT_NOPLASMAREGEN, CONQUEROR_OBLITERATION_TRAIT)
+	particle_holder = new(xeno_owner, /particles/obliteration_warning)
+	particle_holder.alpha = 0
+	particle_holder.layer = ABOVE_NORMAL_TURF_LAYER
+	particle_holder.pixel_x = xeno_owner.pixel_x + 32
+	particle_holder.pixel_y = xeno_owner.pixel_y
+	START_PROCESSING(SSprocessing, src)
+	succeed_activate()
+	RegisterSignal(xeno_owner, COMSIG_MOB_DEATH, PROC_REF(disable_ability))
+	if(keybind && xeno_owner.client)
+		RegisterSignal(xeno_owner.client, COMSIG_XENOABILITY_CONQUEROR_OBLITERATION_UP, PROC_REF(check_ability))
+
+/// Checks if we meet the conditions to begin our attack.
+/datum/action/ability/xeno_action/conqueror_obliteration/proc/check_ability()
+	SIGNAL_HANDLER
+	if(xeno_owner.client)
+		UnregisterSignal(xeno_owner.client, COMSIG_XENOABILITY_CONQUEROR_OBLITERATION_UP)
+	STOP_PROCESSING(SSprocessing, src)
+	if(particle_holder)
```

Comment:
isnt this a more "try_start_attack" than a "check ability"

Reply:
Yeah. I've updated the proc name in the next commit.